### PR TITLE
Add test case for internal/errors/ngt.go

### PR DESCRIPTION
**Pull request by hlts2**
Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

### Description:

I added the test cases for internal/errors/ngt.go and comments and refactored implementation code.

#### refactor
- WHY
The output of following code is `ngt object uuid not found%!(EXTRA uint32=0)` .
https://github.com/vdaas/vald/blob/master/internal/errors/ngt.go#L85

- WHAT

I changed from `Errorf` to `New`
And I searched for code that uses this implementation. 
According to the results of the investigation, this fix had no effect.
https://github.com/vdaas/vald/search?q=ErrUUIDNotFound

The following is a handler that uses `insert` and` delete` in the above result, but it is not affected by the wording correction.

handler of `Insert`
- https://github.com/vdaas/vald/blob/5c914a2fa02fd0b128fbf24933523dd79d1a31ee/pkg/agent/core/ngt/handler/grpc/handler.go#L282

handler of `Delete`
- https://github.com/vdaas/vald/blob/5c914a2fa02fd0b128fbf24933523dd79d1a31ee/pkg/agent/core/ngt/handler/grpc/handler.go#L526

**grammar check passed**

### Environment:

- Go Version: 1.15.6
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/errors/ngt.go [hunk 1]
```diff
@@ -18,83 +18,101 @@
...
 
+	// ErrInvalidDimensionSize represents a function to generate an error that the dimension size is invalid.
 	ErrInvalidDimensionSize = func(current, limit int) error {
 		if limit == 0 {
 			return Errorf("dimension size %d is invalid, the supporting dimension size must be bigger than 2", current)
 		}
 		return Errorf("dimension size %d is invalid, the supporting dimension size must be between 2 ~ %d", current, limit)
 	}
 
+	// ErrDimensionLimitExceed represents a function to generate an error that the supported dimension limit exceded.
```
Comment (hlts2) [inline internal/errors/ngt.go:40]:
```suggestion
	// ErrDimensionLimitExceed represents a function to generate an error that the supported dimension limit exceeded.
```

File: internal/errors/ngt.go [hunk 2]
```diff
@@ -18,83 +18,101 @@
...
 	ErrFailedToSetCreationEdgeSize = func(err error) error {
 		return Wrap(err, "failed to set creation edge size")
 	}
 
+	// ErrFailedToSetSearchEdgeSize represents a function to generate an error that the set of search edge size failed.
 	ErrFailedToSetSearchEdgeSize = func(err error) error {
 		return Wrap(err, "failed to set search edge size")
 	}
 
+	// ErrUncommittedIndexExists represents a function to generate an error that the uncommitted indexes exists.
```
Comment (hlts2) [inline internal/errors/ngt.go:81]:
```suggestion
	// ErrUncommittedIndexExists represents a function to generate an error that the uncommitted indexes exist.
```

File: internal/errors/ngt_test.go [hunk 3]
```diff
@@ -0,0 +1,1243 @@
...
+			args: args{
+				current: 0,
+				limit:   0,
+			},
+			want: want{
+				want: New("supported dimension limit exceed:\trequired = 0,\tlimit = 0"),
+			},
+		},
+		{
+			name: "return ErrDimensionLimitExceed error when current and limit are the minimum value of math.Int64",
```
Comment (hlts2) [inline internal/errors/ngt_test.go:211]:
```suggestion
			name: "return ErrDimensionLimitExceed error when current and limit are the minimum value of int",
```

File: internal/errors/ngt_test.go [hunk 4]
```diff
@@ -0,0 +1,1243 @@
...
+			args: args{
+				current: int(math.MinInt64),
+				limit:   int(math.MinInt64),
+			},
+			want: want{
+				want: Errorf("supported dimension limit exceed:\trequired = %d,\tlimit = %d", int(math.MinInt64), int(math.MinInt64)),
+			},
+		},
+		{
+			name: "return ErrDimensionLimitExceed error when current and limit are the maximum value of math.Int64",
```
Comment (hlts2) [inline internal/errors/ngt_test.go:221]:
```suggestion
			name: "return ErrDimensionLimitExceed error when current and limit are the maximum value of int",
```

File: internal/errors/ngt_test.go [hunk 5]
```diff
@@ -0,0 +1,1243 @@
...
+			name: "return wrapped ErrFailedToSetCreationEdgeSize error when err is ngt error",
+			args: args{
+				err: New("ngt error"),
+			},
+			want: want{
+				want: New("failed to set creation edge size: ngt error"),
+			},
+		},
+		{
+			name: "return ErrFailedToSetCreationEdgeSize error when err is ngt error",
```
Comment (hlts2) [inline internal/errors/ngt_test.go:608]:
```suggestion
			name: "return ErrFailedToSetCreationEdgeSize error when err is nil",
```

File: internal/errors/ngt_test.go [hunk 6]
```diff
@@ -0,0 +1,1243 @@
...
+			name: "return ErrUUIDNotFound error when id is 1234",
+			args: args{
+				id: 1234,
+			},
+			want: want{
+				want: New("ngt object uuid 1234's metadata not found"),
+			},
+		},
+		{
+			name: "return ErrUUIDNotFound error when id is maximum value of math.Uint32",
```
Comment (hlts2) [inline internal/errors/ngt_test.go:988]:
```suggestion
			name: "return ErrUUIDNotFound error when id is the maximum value of uint32",
```

File: internal/errors/ngt_test.go [hunk 7]
```diff
@@ -0,0 +1,1243 @@
...
+				},
+				want: want{
+					want: New("0 indexes are not committed"),
+				},
+			}
+		}(),
+		func() test {
+			var num uint64 = math.MaxUint64
+			return test{
+				name: "return ErrUncommittedIndexExists error when num is the maximum value of math.Uint64",
```
Comment (hlts2) [inline internal/errors/ngt_test.go:748]:
```suggestion
				name: "return ErrUncommittedIndexExists error when num is the maximum value of uint64",
```

File: internal/errors/ngt_test.go [hunk 8]
```diff
@@ -0,0 +1,1243 @@
...
+			args: args{
+				uuid: "",
+				oid:  100,
+			},
+			want: want{
+				want: New("ngt uuid  object id 100 already exists "),
+			},
+		},
+		{
+			name: "return ErrUUIDAlreadyExists error when uuid is '550e8400-e29b-41d4' and oid is maximum value of math.Uint64",
```
Comment (hlts2) [inline internal/errors/ngt_test.go:915]:
```suggestion
			name: "return ErrUUIDAlreadyExists error when uuid is '550e8400-e29b-41d4' and oid is the maximum value of uint64",
```

**Review by vankichi — CHANGES_REQUESTED** [review]
please check these suggestion

File: internal/errors/ngt_test.go [hunk 9]
```diff
@@ -0,0 +1,1337 @@
...
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return ErrUnsupportedDistanceType error",
```
Comment (hlts2) [inline internal/errors/ngt_test.go:418]:
```suggestion
			name: "return an ErrUnsupportedDistanceType error",
```

File: internal/errors/ngt_test.go [hunk 10]
```diff
@@ -0,0 +1,1337 @@
...
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return wrapped ErrFailedToSetDimension error when err is ngt error",
```
Comment (hlts2) [inline internal/errors/ngt_test.go:638]:
```suggestion
			name: "return a wrapped ErrFailedToSetDimension error when err is ngt error",
```

File: internal/errors/ngt_test.go [hunk 11]
```diff
@@ -0,0 +1,1337 @@
...
+			args: args{
+				err: New("ngt error"),
+			},
+			want: want{
+				want: New("failed to set dimension: ngt error"),
+			},
+		},
+		{
+			name: "return ErrFailedToSetDimension error when err is nil",
```
Comment (hlts2) [inline internal/errors/ngt_test.go:647]:
```suggestion
			name: "return an ErrFailedToSetDimension error when err is nil",
```

File: internal/errors/ngt_test.go [hunk 12]
```diff
@@ -0,0 +1,1337 @@
...
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return ErrUncommittedIndexNotFound error",
```
Comment (hlts2) [inline internal/errors/ngt_test.go:892]:
```suggestion
			name: "return an ErrUncommittedIndexNotFound error",
```

File: internal/errors/ngt_test.go [hunk 13]
```diff
@@ -0,0 +1,1337 @@
...
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return ErrCAPINotImplemented error",
```
Comment (hlts2) [inline internal/errors/ngt_test.go:938]:
```suggestion
			name: "return an ErrCAPINotImplemented error",
```

File: internal/errors/ngt_test.go [hunk 14]
```diff
@@ -0,0 +1,1337 @@
...
+			args: args{
+				err: New("ngt error"),
+				t:   "",
+			},
+			want: want{
+				want: New("failed to set object type : ngt error"),
+			},
+		},
+		{
+			name: "return an ErrFailedToSetObjectType error when err is nil and t is Float",
```
Comment (hlts2) [inline internal/errors/ngt_test.go:574]:
```suggestion
			name: "return an ErrFailedToSetObjectType error when err is nil and t is Int",
```

**Review by vankichi — APPROVED** [review]

**Review by kevindiu — DISMISSED** [review]
LGTM

**Pull request merged**